### PR TITLE
refactor: use PUT on `/position` to move authenticator

### PR DESCRIPTION
**Pull request by sstrigler**
Fixes [EMQX-7863](https://emqx.atlassian.net/browse/EMQX-7863)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] ~For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)~
- [x] For internal contributor: there is a jira ticket to track this change
- [x] ~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~
- [x] ~In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section~
- [x] Sync with dashboard team on changes

### CODE REVIEW

File: apps/emqx_authn/src/emqx_authn_api.erl [hunk 1]
```diff
@@ -99,14 +98,14 @@ paths() ->
         "/authentication",
         "/authentication/:id",
         "/authentication/:id/status",
-        "/authentication/:id/move",
```
Comment (HJianBo) [inline apps/emqx_authn/src/emqx_authn_api.erl:102]: nit: Do we need to be compatible with the old path of the API?

Comment (sstrigler) [reply apps/emqx_authn/src/emqx_authn_api.erl:102]: Yeah, I'm trying to find out

Comment (sstrigler) [reply apps/emqx_authn/src/emqx_authn_api.erl:102]: I synched with @ysfscream - the plan is to update the dashboard in time before the next release, i.e. get this one merged and they'll fix the dashboard in time

File: apps/emqx_authn/src/emqx_authn_api.erl [hunk 2]
```diff
@@ -321,35 +317,27 @@ schema("/listeners/:listener_id/authentication/:id/status") ->
...
+schema("/listeners/:listener_id/authentication/:id/position/:position") ->
     #{
-        'operationId' => listener_authenticator_move,
-        post => #{
+        'operationId' => listener_authenticator_position,
+        put => #{
             tags => ?API_TAGS_SINGLE,
-            description => ?DESC(listeners_listener_id_authentication_id_move_post),
-            parameters => [param_listener_id(), param_auth_id()],
-            'requestBody' => emqx_dashboard_swagger:schema_with_examples(
```
Comment (savonarola) [inline apps/emqx_authn/src/emqx_authn_api.erl:349]: Seems we removed body validation here.

Comment (savonarola) [reply apps/emqx_authn/src/emqx_authn_api.erl:349]: Nwm, I see, the body is not used here.

**Review by savonarola — DISMISSED** [review]

File: changes/v5.0.12-zh.md [hunk 3]
```diff
@@ -4,6 +4,9 @@
 
 - 通过 `node.global_gc_interval = disabled` 来禁用全局垃圾回收 [#9418](https://github.com/emqx/emqx/pull/9418)。
 
+- [FIXME] Refactor authn API by replacing `POST /authentication/{id}/move` with `PUT /authentication/{id}/position/{position}`. [#9419](https://github.com/emqx/emqx/pull/9419)
```
Comment (sstrigler) [inline changes/v5.0.12-zh.md:7]: @lafirest please help!

**Comment by zmstone** [comment]
will this affect dashboard?

**Comment by sstrigler** [comment]
> will this affect dashboard?

Yes, @ysfscream has been notified about changes

**Comment by ysfscream** [comment]
> will this affect dashboard?

YES. @zmstone If it's merged. We will fix it immediately.

**Review by thalesmg — DISMISSED** [review]

**Review by thalesmg — APPROVED** [review]

**Pull request merged**